Dockerfile

```diff
@@ -1,6 +1,8 @@
 FROM openjdk:17-jdk-slim
+ARG build_type=
+ENV MODE=${build_type}
 
 EXPOSE 8080
 ADD /target/*.jar demo.jar
 
-ENTRYPOINT ["java","-jar", "-Dspring.profiles.active=prod" ,"demo.jar"]
+ENTRYPOINT ["java","-jar", "-Dspring.profiles.active=${MODE}" ,"demo.jar"]
```

Comment:
Did you test this locally? According to this [starkoverflow article](https://stackoverflow.com/questions/37904682/how-do-i-use-docker-environment-variable-in-entrypoint-array) it is possible that the environment variable is not interpolated when the container starts.

Reply:
There were no issues testing locally.